example/provider-extensions/seed/seed-config.yaml.tmpl

```diff
@@ -0,0 +1,16 @@
+# mandatory in all scenarios
```

Comment:
In `seed-config.yaml` users define how their seed should be created. This includes
- deciding if some seed properties should be determined automatically via shoot-info config map.
- deciding if Gardener functions like DNS and Certificate registrations should be used.
Those decisions are needed because we would like to be able to support non Gardener clusters as seeds and there are no properties in gardenlet values.yaml for that.

The minimal required properties to promote a non Gardener cluster to a seed are enlisted here too, that users can create their initial configuration in one place. Starting by configuring an entire gardenlet values.yaml might be overwhelming for beginners. If you are no beginner anymore, you could specify your configuration in `example/provider-extensions/gardenlet/values.yaml` in a more detailed way.

Reply:
I agree with @timuthy that another abstraction doesn't help too much and rather complicates the entire situation. Reading below file, I would probably not feel too good as newbie because I wouldn't be too sure what to enter as `registryDomain`, `relayDomain`, etc.
In this light, just using a `values.yaml` template with the minimal configuration to get started is not worse.

> Those decisions are needed because we would like to be able to support non Gardener clusters as seeds and there are no properties in gardenlet values.yaml for that.

IIUC, this information is extract by a bash script anyways. Can't we keep it easy and always extract the needed values from there if the `shoot-info` `ConfigMap` exists? I can't think of a scenario where I would not want this automatic configuration in case my seed is a shoot cluster.

> Starting by configuring an entire gardenlet values.yaml might be overwhelming for beginners.

As said above, the current file is not simplifying things too much either. We don't need to expose the entire `SeedSpec` in this `values.yaml` but just the minimal fields. As a beginner, I would anyways not start with this "real infrastructure extension" setup but rather get my feet wet with `provider-local`. By the time I switched to this setup, I would probably be at least somewhat familiar with the specification of a `Seed`.

Reply:
Nice, thank you! Can you please also adjust the doc https://github.com/gardener/gardener/pull/6678/files#diff-8a7ddfd3289d874ea3659a508aebfe5b97f2eba63b1bef7b22b58e25db7a0d11R47.

Reply:
docs updated 📚